FreeRTOS_Sockets.c

```diff
@@ -386,8 +386,29 @@ Socket_t FreeRTOS_socket( BaseType_t xDomain,
     size_t uxSocketSize = 1;
     EventGroupHandle_t xEventGroup;
     Socket_t xReturn;
+    BaseType_t xProtocolCpy = xProtocol;
 
-    if( prvDetermineSocketSize( xDomain, xType, xProtocol, &uxSocketSize ) == pdFAIL )
+    /* The special protocol FREERTOS_SOCK_DEPENDENT_PROTO, which is equivalent
+     * to passing 0 as defined by POSIX, indicates to the socket layer that it
+     * should pick a sensible default protocol based off the given socket type.
+     * If we can't, prvDetermineSocketSize will catch it as an invalid
+     * type/protocol combo.
+     */
+    if( xProtocol == FREERTOS_SOCK_DEPENDENT_PROTO )
+    {
+        switch( xType )
+        {
+            case FREERTOS_SOCK_DGRAM:
+                xProtocolCpy = FREERTOS_IPPROTO_UDP;
+                break;
+
+            case FREERTOS_SOCK_STREAM:
+                xProtocolCpy = FREERTOS_IPPROTO_TCP;
+                break;
```

Comment:
One more comment here. Apologies that I did not catch this earlier and it is taking a bit long to get this PR merged.
MISRA rule 16.4 says that all switch statements must have a `default` label.
In this case, I think we can get away with just adding a comment because an incorrect `xType` will be caught by `prvDetermineSocketSize`.

Reply:
Yeah, I should have thought as much. Sorry, I should have read through some of these MISRA rules it seems. Didn't really think that much could be wrong with 10 lines of code :D

Anyways, I fixed the issue. Hopefully everything is in order now.